echopype/clean/api.py

```diff
@@ -64,34 +77,87 @@ def mask_transient_noise(
+    >>>     depth_bin = "10m",
+    >>>     num_side_pings = 25,
+    >>>     exclude_above = "250.0m",
+    >>>     transient_noise_threshold = "12.0dB",
+    >>>     range_var = "depth",
+    >>>     use_index_binning = False
+    >>> )
+
+    Vertical Index Binning Transient Noise Mask Usage with Chunking across Vertical
+    and Horizontal Axis:
+
+    >>> mask_transient_noise(
+    >>>     ds_Sv
+    >>>     func="nanmean",
+    >>>     depth_bin = "10m",
+    >>>     num_side_pings = 25,
+    >>>     exclude_above = "250.0m",
+    >>>     transient_noise_threshold = "12.0dB",
+    >>>     range_var = "depth",
+    >>>     use_index_binning = True,
+    >>>     chunk_dict = {"ping_time": 1000, "range_sample": 50}
+    >>> )
+
+    To compute the pooled Sv, prior to the binning operation, the number of range sample
+    indices needed to encapsulate "10m" will be calculated, and this number will be used to
+    do a completely index based nanmean binning operation. This operation is exactly the
+    same as applying a mean filter over an image, but in this case, it is applied to an
+    Echogram.
     """
-    if "depth" not in ds_Sv.data_vars:
+    # Check range variable
+    if range_var not in ["echo_range", "depth"]:
+        raise ValueError("`range_var` must be either `echo_range` or `depth`.")
+    if range_var not in ds_Sv.data_vars and not use_index_binning:
         raise ValueError(
-            "Masking attenuated signal requires depth data variable in `ds_Sv`. "
-            "Consider adding depth with `ds_Sv = ep.consolidate.add_depth(ds_Sv)`."
+            f"Masking transient noise requires `{range_var}` data variable in `ds_Sv`."
         )
 
     # Copy `ds_Sv`
     ds_Sv = ds_Sv.copy()
 
     # Check for appropriate function passed in
     if func != "nanmean" and func != "nanmedian":
-        raise ValueError("Input `func` is `nanmode`. `func` must be `nanmean` or `nanmedian`.")
-
-    # Warn when `func=nanmedian` since the sorting overhead makes it incredibly slow compared to
-    # `nanmean`.
-    logger.warning(
-        "Consider using `func=nanmean`. `func=nanmedian` is an incredibly slow operation due to "
-        "the overhead sorting."
-    )
-
-    # Setup depth bin
-    ds_Sv, depth_bin = _setup_and_validate(ds_Sv, "depth", depth_bin)
-
-    # Compute pooled Sv
-    pooled_Sv = calc_transient_noise_pooled_Sv(
-        ds_Sv, func, depth_bin, num_side_pings, exclude_above
-    )
+        raise ValueError(f"Input `func` is `{func}`. `func` must be `nanmean` or `nanmedian`.")
+    # Assign string to numpy function
+    if func == "nanmean":
+        func = np.nanmean
+    elif func == "nanmedian":
+        # Warn when `func=nanmedian` since the sorting overhead makes it incredibly slow compared to
+        # `nanmean`.
+        logger.warning(
+            "Consider using `func=nanmean`. `func=nanmedian` is an incredibly slow operation due "
+            "to the overhead sorting."
+        )
```

Comment:
I agree with the statement, but the original method used `nanmedian` right? I feel that we should not recommend the change, but we can warn people that it is SLOW. We can also point them to the issue for the Fielding filter and the echopy code for that in the warning?

Reply:
Added this. Good idea! 👍